main.tf

```diff
@@ -33,6 +33,395 @@ locals {
+  wait                       = try(var.cloudwatchlogs.wait, false)
+  wait_for_jobs              = try(var.cloudwatchlogs.wait_for_jobs, null)
+  dependency_update          = try(var.cloudwatchlogs.dependency_update, null)
+  replace                    = try(var.cloudwatchlogs.replace, null)
+  lint                       = try(var.cloudwatchlogs.lint, null)
+
+  postrender = try(var.cloudwatchlogs.postrender, [])
+
+  set = concat([
+    {
+      # shortens pod name from `ack-cloudwatchlogs-cloudwatchlogs-chart-xxxxxxxxxxxxx` to `ack-cloudwatchlogs-xxxxxxxxxxxxx`
+      name  = "nameOverride"
+      value = "ack-cloudwatchlogs"
+    },
+    {
+      name  = "aws.region"
+      value = local.region
+    },
+    {
+      name  = "serviceAccount.name"
+      value = local.cloudwatchlogs_name
+    }],
+    try(var.cloudwatchlogs.set, [])
+  )
+  set_sensitive = try(var.cloudwatchlogs.set_sensitive, [])
+
+  # IAM role for service account (IRSA)
+  set_irsa_names                = ["serviceAccount.annotations.eks\\.amazonaws\\.com/role-arn"]
+  create_role                   = try(var.cloudwatchlogs.create_role, true)
+  role_name                     = try(var.cloudwatchlogs.role_name, "ack-cloudwatchlogs")
+  role_name_use_prefix          = try(var.cloudwatchlogs.role_name_use_prefix, true)
+  role_path                     = try(var.cloudwatchlogs.role_path, "/")
+  role_permissions_boundary_arn = lookup(var.cloudwatchlogs, "role_permissions_boundary_arn", null)
+  role_description              = try(var.cloudwatchlogs.role_description, "IRSA for CloudWatch Logs controller for ACK")
+  role_policies = lookup(var.cloudwatchlogs, "role_policies", {
+    policy = var.enable_cloudwatchlogs ? aws_iam_policy.cloudwatchlogs[0].arn : null
+  })
+  create_policy = try(var.cloudwatchlogs.create_policy, false)
+
+  oidc_providers = {
+    this = {
+      provider_arn = local.oidc_provider_arn
+      # namespace is inherited from chart
+      service_account = local.cloudwatchlogs_name
+    }
+  }
+
+  tags = var.tags
+}
+
+# recommended cloudwatchlogs-controller policy https://github.com/aws-controllers-k8s/cloudwatchlogs-controller/blob/main/config/iam/recommended-inline-policy
+data "aws_iam_policy_document" "cloudwatchlogs" {
+  count = var.enable_cloudwatchlogs ? 1 : 0
+
+  statement {
+    sid    = "VisualEditor0"
+    effect = "Allow"
+
+    actions = [
+      "logs:TagLogGroup",
+      "logs:DescribeLogGroups",
+      "logs:UntagLogGroup",
+      "logs:DeleteLogGroup",
+      "logs:UntagResource",
+      "logs:TagResource",
+      "logs:CreateLogGroup",
+      "logs:ListTagsForResource",
+    ]
+
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "cloudwatchlogs" {
+  count = var.enable_cloudwatchlogs ? 1 : 0
+
+  name        = "CloudWatchLogsController"
+  description = "IAM policy for CloudWatch Logs Controller"
+  policy      = data.aws_iam_policy_document.cloudwatchlogs[0].json
+
+  tags = var.tags
+}
```

Comment:
Instead of creating a resource for the policy, attach the json directly to `source_policy_document`, same as the [addons repo](https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/blob/c003cad83f930a16395a5e4cddd7daa1d74c76a3/main.tf#L1474).